### PR TITLE
DB caching/Request reduction features. 

### LINKED ISSUE #2: Rate limiting causing images not to be displayed in frontend
This is due to rate limiting on google places API. This could be due to the amount of images we are requesting. The Max is 10 that is outputted per place_id. 

We should look at limiting this to 4-5 images per request.

We also need to move the request to be a passthrough from our API, so the frontend is making the request to our API, which then passes the image file sent by Google Places through to our site.

### LINKED ISSUE #5: DB check: cannot read place_id of undefined
This is part of a bigger issue relating to error handling for our API.

We will need to construct some error handling for things like malformed requests, timeouts, responses from other APIs and bugs. Making sure to try and account for expected operational errors.

https://www.toptal.com/nodejs/node-js-error-handling

This article runs through how to build classes to handle errors and responses through an API, on errors calling the class to parse through the error to determine the appropriate response.

We will need to do the same, build one class to parse through errors then call a response handler to respond to the request.

### PR DESCRIPTION
**Pull request by Gerrudo**
Testing for DB caching feature complete.

Known issues with error handling, a new branch will be created after this PR to implement classes for error handling and response.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -65,122 +61,161 @@ function reusableRequest(url, method){
...
+        });
+    });
+};
+
 /*
 API HANDLERS
 */
 
 //Handler for /go when userCoordinates is provided
 app.post('/newlocationsearch', (req, res) => {
```
Comment (kieran-flaherty) [inline server.js]: Just having a look at these handler functions here, they just call into another function, which doesn't really make sense. What might be better is to pull the try/catch block up a level, and then you can catch any errors and handle them, usually logging the error somewhere. You can also decide which response code to send back to the requester. It's probably not best practice to just return the error either, as we want to obsure as much of the back end from the FE as possible, maybe return a custom error object something like `{ "message":"There was an error, please try again", "status_code":500 }`

File: server.js [hunk 2]
```diff
@@ -36,20 +36,16 @@ MongoDB Connectors/Collection creation
 */
 
 const MongoClient = require('mongodb').MongoClient;
-const dbName = 'restaurants-web-api-db'
-const dbHost = `protainer.lan:27017`
-const url = `mongodb://${dbHost}/${dbName}`;
 
-MongoClient.connect(url, function(err, db) {
+MongoClient.connect(dbConfig.url, function(err, db) {
```
Comment (kieran-flaherty) [inline server.js]: So there is a lot of code in this file now. As I imagine you want to add more features, now is probably a good time to start separating our concerns and splitting things out into different classes/files before the project gets bigger. One good place to start might be the code to access the database. At the minute we have one collection called recentlocations, so maybe create a file called recentlocationsrepository, and in here we can have a mongo client, then split out each db call into a function that we can import for use in another file. Something a bit like:
```
const MongoClient = require('mongodb').MongoClient;

function checkMongoClientConnection() {
  MongoClient.connect(dbConfig.url, function(err, db) {
    if (err) console.error(err);
    db.close();
  });
}

function initialiseRecentLocationsCollection() {
  MongoClient.connect(dbConfig.url, function(err, db) {
      if (err) console.error(err);
      let dbo = db.db(dbConfig.dbName);
      dbo.createCollection("recentlocations", function(err, res) {
          db.close();
      });
  });
}
```

File: server.js (hunk 1 above)
Comment (Gerrudo) [reply server.js]: Please check commit cced6f1dceea37573de7b01d2035013f24ed50e4, I've moved the try catch into the handler, which will now successfully respond based on responses from other APIs and any other request/DB errors. Also sends 500 with customer error json in the response.

File: server.js (hunk 2 above)
Comment (Gerrudo) [reply server.js]: Please see InitaliseMongoCollections.js in 727882ab114e81b1cb94d3f7fdb1fc2a8569cb35

I've split that into their own file, could go further but that seems enough for now. Probably will look at turning other other 2 files into Classes, let me know what you think.

**Pull request merged**